### PR TITLE
Dragging a field while unfocused should do nothing

### LINKED ISSUE #93143: TextField cannot be unfocused when mouse primary button is pressed
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Select text in the text field and keep the primary button pressed
3. Move cursor outside of the container to trigger the unfocusing callback

**Expected results:** 

localhost:60694:#: 2021-11-05 19-18-44

**Actual results:** 

https://user-images.githubusercontent.com/13119003/140543911-03e576de-e19e-4d24-87bb-9c134dce9ed3.mp4

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatefulWidget {
  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final _focusNode = FocusNode();
  final _controller = TextEditingController();

  @override
  void dispose() {
    _focusNode.dispose();
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: SafeArea(
          child: Column(
            children: [
              MouseRegion(
                onExit: (_) {
                  _focusNode.unfocus();
                },
                child: Container(
                  height: 200,
                  width: 200,
                  color: Colors.green.withOpacity(0.1),
                  child: Center(
                    child: TextField(
                      focusNode: _focusNode,
                      controller: _controller,
                      decoration: InputDecoration(border: OutlineInputBorder()),
                    ),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-x64, locale en-GB)
    • Flutter version 2.5.3 at /Users/krasnov/sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (3 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/krasnov/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.10.0.beta.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 95.0.4638.69

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by justinmc**
Fixes https://github.com/flutter/flutter/issues/93143

### CODE REVIEW

**Comment by justinmc** [comment]
I'm struggling to reproduce the test failures locally.  Pushed a merge commit.

**Comment by goderbauer** [comment]
(triage) @justinmc Is this one still on your radar?

**Comment by justinmc** [comment]
Now it is, thanks for the reminder!  I think I'm going to be able to fix the tests and get this into review.

File: packages/flutter/lib/src/cupertino/text_field.dart [hunk 1]
```diff
@@ -125,6 +125,14 @@ class _CupertinoTextFieldSelectionGestureDetectorBuilder extends TextSelectionGe
     _state._requestKeyboard();
     super.onDragSelectionEnd(details);
   }
+
+  @override
+  void onDragSelectionUpdate(TapDragUpdateDetails details) {
+    if (!_state._effectiveFocusNode.hasFocus) {
```
Comment (justinmc) [inline packages/flutter/lib/src/cupertino/text_field.dart:131]: This works in practice but not in some tests below (without autofocusing the field first).  In onDragSelectionStart, the field is focused with `requestFocus`, but that may take an additional frame to be set in `hasFocus`.  In the tests, onDragSelectionUpdate is called before `hasFocus` is true.

**Comment by justinmc** [comment]
I'm struggling to test this, but I also should verify that this is behavior that we want for Flutter by default.  I should try something similar to the example in the issue but in plain HTML/JS and see what happens (and maybe on native Android and iOS too).

Also @Renzo-Olivares may come up with a way to disable dragging in the future, since it interferes with Scrollables sometimes.

**Comment by justinmc** [comment]
Closing since https://github.com/flutter/flutter/issues/93143 is closed.

**Pull request closed without merging**